### PR TITLE
576 bug fix baseline cewl

### PR DESCRIPTION
**Pull request by anjojoseph**
Hi Team,
Kindly review and approve PR for merging baseline fix - cewl tool.
Thank you.

### CODE REVIEW

File: src/components/Cewl/Cewl.tsx [hunk 1]
```diff
@@ -162,121 +234,138 @@ const Cewl = () => {
...
+                            />
+                            <Checkbox
+                                label={"Include and show email addresses"}
+                                checked={checkedEmail}
+                                onChange={(e) => setCheckedEmail(e.currentTarget.checked)}
+                            />
+                        </>
+                    )}
+                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
+                    <Button type={"submit"}>Scan</Button>
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:364]: The format for baselining is Start <title of tool> 
```suggestion
                    <Button type={"submit"}>Start {title}</Button>
```

File: src/components/Cewl/Cewl.tsx [hunk 2]
```diff
@@ -29,22 +19,50 @@ interface FormValuesType {
...
+        "The tool Cewl, renown for being a custom word list generator, is a ruby app which spiders given URLs to " +
+        "a specified depth to return a list of words that are able to be used within password crackers including " +
+        " JohnTheRipper (which can be found within the DDT). This tool is particularly useful for security testing and " +
+        "forensic investigation.\nOptions for the tool can be found at:  https://www.kali.org/tools/cewl/#:~:text=CeWL" +
+        "%20(Custom%20Word%20List%20generator,\nCeWL%20can%20follow%20external%20links."; // Description of the component.
+    const steps =
+        "Step 1: Enter the maximum depth to spider to. Eg: 2\n" +
+        "Step 2: Enter a minimum word length. Eg: 3\n" +
+        "Step 3: Enter a target URL. Eg: google.com\n" +
+        "Step 4: Click 'Scan' to commence Cewl's operation.\n" +
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:59]: Change scan to Start <title> 
```suggestion
        "Step 4: Click 'Start " + title + "' to commence Cewl's operation.\n" +
```

File: src/components/Cewl/Cewl.tsx [hunk 3]
```diff
@@ -29,22 +19,50 @@ interface FormValuesType {
...
+        "Step 5: View the output block below to view the results of the tools execution.";
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:60]: Needs an apostrophe. 
```suggestion
        "Step 5: View the output block below to view the results of the tool's execution.";
```

File: src/components/Cewl/Cewl.tsx [hunk 4]
```diff
@@ -29,22 +19,50 @@ interface FormValuesType {
...
+    const sourceLink = "https://salsa.debian.org/pkg-security-team/cewl"; // Link to the source code (or Kali Tools).
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["ruby"]; // Contains the dependencies required by the component.
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:63]: Need to also check the cewl itself is installed. 
```suggestion
    const dependencies = ["ruby", "cewl"]; // Contains the dependencies required by the component.
```

File: src/components/Cewl/Cewl.tsx [hunk 5]
```diff
@@ -162,121 +234,138 @@ const Cewl = () => {
...
-                {checkedAdvanced && (
-                    <>
-                        <TextInput
-                            label={"Max depth"}
-                            type="number"
-                            placeholder={"Deafult is set to 2"}
-                            {...form.getInputProps("depth")}
-                        />
-                        <div style={{ display: "flex", alignItems: "center" }}>
+
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:254]: Delete empty line. 
```suggestion
```

File: src/components/Cewl/Cewl.tsx [hunk 6]
```diff
@@ -162,121 +234,138 @@ const Cewl = () => {
...
+                        label={"Save wordList"}
+                        placeholder={"/home/kali/Desktop/WordList.txt"}
+                        {...form.getInputProps("wordList")}
+                    />
+                    {checkedAdvanced && (
+                        <>
                             <TextInput
-                                label={"Min Length"}
-                                placeholder={"0"}
+                                label={"Max depth"}
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:285]: There is inconsistency with the use of capitalisation of the 2nd word. For example, should it be "Max Depth" or "Max depth"? For the next field you use capital "L" for "Min Length".

File: src/components/Cewl/Cewl.tsx [hunk 7]
```diff
@@ -162,121 +234,138 @@ const Cewl = () => {
...
-                        />
-                    </>
-                )}
-                {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
-                <Button type={"submit"}>Scan</Button>
-                <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
-            </Stack>
-        </form>
+                            <Checkbox
+                                label={"Show the count for earch word found"}
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:352]: Typo. 
```suggestion
                                label={"Show the count for each word found"}
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @anjojoseph 

Thank you this is looking good. I have recommended some minor changes. 

I tried testing the basic version of this tool, but the wordlist did not save on my local machine. 

Were you able to get the tool working? 

Kind regards

Michael

File: src/components/Cewl/Cewl.tsx [hunk 8]
```diff
@@ -162,121 +234,138 @@ const Cewl = () => {
...
-                        <div style={{ display: "flex", alignItems: "center" }}>
+
+            <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
+                <LoadingOverlay visible={loading} />
+                {loading && (
+                    <div>
+                        <Button variant="outline" color="red" style={{ zIndex: 1001 }} onClick={handleCancel}>
+                            Cancel
+                        </Button>
+                    </div>
```
[the remark this answers is not in the document]
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:262]: updated

File: src/components/Cewl/Cewl.tsx [hunk 9]
```diff
@@ -29,22 +19,50 @@ interface FormValuesType {
...
+    const [checkedEmail, setCheckedEmail] = useState(false); // State variable to include and show email addresses.
+    const [checkedLowercase, setCheckedLowercase] = useState(false); // State variable to show only lowercase words.
 
+    // Component Constants.
+    const methods = ["No authentication", "Basic", "Digest"]; // Variable to store authentication methods.
+    const title = "Cewl"; // Title of the component.
+    const description =
+        "The tool Cewl, renown for being a custom word list generator, is a ruby app which spiders given URLs to " +
+        "a specified depth to return a list of words that are able to be used within password crackers including " +
+        " JohnTheRipper (which can be found within the DDT). This tool is particularly useful for security testing and " +
```
[the remark this answers is not in the document]
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:52]: corrected

File: src/components/Cewl/Cewl.tsx [hunk 10]
```diff
@@ -29,22 +19,50 @@ interface FormValuesType {
...
+        "forensic investigation.\nOptions for the tool can be found at:  https://www.kali.org/tools/cewl/#:~:text=CeWL" +
```
[the remark this answers is not in the document]
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:53]: done

File: src/components/Cewl/Cewl.tsx [hunk 11]
```diff
@@ -29,22 +19,50 @@ interface FormValuesType {
...
+        "%20(Custom%20Word%20List%20generator,\nCeWL%20can%20follow%20external%20links."; // Description of the component.
```
[the remark this answers is not in the document]
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:54]: removed

File: src/components/Cewl/Cewl.tsx (hunk 3 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:60]: done

File: src/components/Cewl/Cewl.tsx (hunk 5 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:254]: done

File: src/components/Cewl/Cewl.tsx (hunk 4 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:63]: done

File: src/components/Cewl/Cewl.tsx (hunk 6 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:285]: done

File: src/components/Cewl/Cewl.tsx (hunk 7 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:352]: done

File: src/components/Cewl/Cewl.tsx (hunk 2 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx:59]: done

**Comment by anjojoseph** [comment]
Hi Michael/Team,
I have added all suggested changes.
Kindly review and approve.
The tool works and created wordlist for me and the updated cancel button also work as expected.
uploading screenshots for your reference kindly check.

image


image

File: src/components/Cewl/Cewl.tsx (hunk 2 above)
Comment (michaeljpigott) [reply src/components/Cewl/Cewl.tsx:59]: I think you might have missed this one (?) 🤔

File: src/components/Cewl/Cewl.tsx [hunk 12]
```diff
@@ -29,22 +20,49 @@ interface FormValuesType {
...
+    const description =
+        "The tool Cewl, renown for being a custom word list generator, is a ruby app which spiders given URLs to " +
+        "a specified depth to return a list of words that are able to be used within password crackers including " +
+        " JohnTheRipper (which can be found within DDT). This tool is particularly useful for security testing and " +
+        "forensic investigation.\nOptions for the tool can be found at:  https://www.kali.org/tools/cewl/"; // Description of the component.
+    const steps =
+        "Step 1: Enter the maximum depth to spider to. Eg: 2\n" +
+        "Step 2: Enter a minimum word length. Eg: 3\n" +
+        "Step 3: Enter a target URL. Eg: google.com\n" +
+        "Step 4: Click 'Scan' to commence Cewl's operation.\n" +
```
Comment (michaeljpigott) [inline src/components/Cewl/Cewl.tsx:59]: Need to change this to Start plus the tool title. 
```suggestion
        "Step 4: Click Start " + title + " to commence Cewl's operation.\n" +
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @anjojoseph 

Good work. This is just about done. I think you may have missed the adjustment to Step 4 in the user guide, replacing "Scan" with "Start Cewl". 

I have tested the tool and it works.

File: src/components/Cewl/Cewl.tsx (hunk 12 above)
Comment (anjojoseph) [reply src/components/Cewl/Cewl.tsx]: updated

**Comment by anjojoseph** [comment]
Hi Michael,
updated the step.

**Review by michaeljpigott — APPROVED** [review]
Hi @anjojoseph 

Thank you making that change. I have approved your pull request. Please do not forget to close the corresponding Planner card. 

Kind regards

Michael

**Pull request merged**